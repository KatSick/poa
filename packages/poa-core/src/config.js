// @ts-check
import React from 'react';

function helloComponent() {
  return <div>Jambo, jambo! Define your routes in {`boot({ routes: <here> })`}.</div>;
}

export function createDefaultConfig(config) {
  const defaultConfig = {
    react: { htmlNode: document.getElementById('root'), loadingComponent: null },
    router: {
      type: 'browser',
      routes: [
        {
          path: '',
          component: helloComponent
        }
      ]
    },
    state: {
      initial: {},
      actions: {}
    },
    env: {}
  };

  return Object.assign({}, defaultConfig, config);
}
