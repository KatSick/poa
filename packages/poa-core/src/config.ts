import { PoaAppConfig, PoaRouterType } from './interfaces/app-config.interface';
import { helloComponent } from './hello.component';

export function createDefaultConfig(config?: PoaAppConfig) {
  const defaultConfig = {
    react: { htmlNode: document.getElementById('root'), loadingComponent: null },
    router: {
      type: PoaRouterType.Browser,
      routes: [
        {
          path: '',
          component: helloComponent
        }
      ]
    },
    state: {
      initial: {},
      actions: {}
    },
    i18n: {},
    env: {}
  };

  return Object.assign({}, defaultConfig, config);
}
