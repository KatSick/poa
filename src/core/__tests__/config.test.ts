import { createDefaultConfig } from '../config';
import { PoaAppBootConfig, PoaRouterType } from '../interfaces/app-config.interface';

describe('CORE â€” Config', () => {
  it(`creates config without arguments`, () => {
    expect(createDefaultConfig()).toMatchSnapshot();
  });

  it(`merges defaultConfig with condig provided as argument`, () => {
    const config: PoaAppBootConfig = {
      env: { test: {} },
      hooks: {
        configureAppInstance: () => Promise.resolve(1),
        configureI18Next: () => Promise.resolve(2)
      },
      i18n: {},
      react: { htmlNode: {} as any, loadingComponent: {} as any },
      router: { type: PoaRouterType.Memory, routes: [] },
      state: { initial: { test: 'value' }, actions: { the: 'action' } }
    };

    expect(createDefaultConfig(config)).toMatchSnapshot();
    expect(createDefaultConfig(config)).toEqual(config);
  });

  it(`merges deep objects`, () => {
    const config: PoaAppBootConfig = {
      env: { test: 'value' }
    };

    expect(createDefaultConfig(config).env).toHaveProperty('te123st');
    // expect(createDefaultConfig(config)).toEqual(config);
  });
});
