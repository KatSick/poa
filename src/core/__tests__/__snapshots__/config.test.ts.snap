// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CORE — Config creates config without arguments 1`] = `
Object {
  "env": Object {},
  "hooks": Object {
    "configureAppInstance": [Function],
    "configureI18Next": [Function],
  },
  "i18n": Object {},
  "react": Object {
    "htmlNode": null,
    "loadingComponent": null,
  },
  "router": Object {
    "routes": Array [
      Object {
        "component": [Function],
        "path": "",
      },
    ],
    "type": "browser",
  },
  "state": Object {
    "actions": Object {},
    "initial": Object {},
  },
}
`;

exports[`CORE — Config merges defaultConfig with condig provided as argument 1`] = `
Object {
  "env": Object {
    "test": Object {},
  },
  "hooks": Object {
    "configureAppInstance": [Function],
    "configureI18Next": [Function],
  },
  "i18n": Object {},
  "react": Object {
    "htmlNode": Object {},
    "loadingComponent": Object {},
  },
  "router": Object {
    "routes": Array [],
    "type": "memory",
  },
  "state": Object {
    "actions": Object {
      "the": "action",
    },
    "initial": Object {
      "test": "value",
    },
  },
}
`;
